src/Header.tsx

```diff
@@ -104,4 +105,9 @@ Header.Link.propTypes = {
   ...sx.propTypes
 }
 
-export default Header
+HeaderLink.defaultProps = {
+  theme
+}
+
+export type HeaderProps = ComponentProps<typeof Header>
+export default Object.assign(Header, {Link: HeaderLink, Item: HeaderItem})
```

Comment:
We also need to export prop types for the sub components:
```suggestion
export type HeaderProps = ComponentProps<typeof Header>
export type HeaderLinkProps = ComponentProps<typeof HeaderLink>
export type HeaderItemProps = ComponentProps<typeof HeaderItem>
export default Object.assign(Header, {Link: HeaderLink, Item: HeaderItem})
```